kong/dao/schemas/apis.lua

```diff
@@ -5,12 +5,14 @@ local fmt = string.format
 local sub = string.sub
 local match = string.match
 
+local PROTOCOLS = { "HTTP", "HTTPS" }
+
 local function validate_upstream_url_protocol(value)
   local parsed_url = url.parse(value)
   if parsed_url.scheme and parsed_url.host then
-    parsed_url.scheme = parsed_url.scheme:lower()
-    if not (parsed_url.scheme == "http" or parsed_url.scheme == "https") then
-      return false, "Supported protocols are HTTP and HTTPS"
+    parsed_url.scheme = parsed_url.scheme:upper()
+    if not utils.table_contains(PROTOCOLS, parsed_url.scheme) then
+      return false, "Supported protocols are: "..table.concat(PROTOCOLS, ", ")
```

Comment:
please refrain from using `table_contains`, it should not even exist. Just create a map.

```lua
local PROTOCOLS = { "HTTP", "HTTPS" }
for i, v in ipairs(PROTOCOLS) do PROTOCOLS[v] = i end

local function validate_upstream_url_protocol(value)
  local parsed_url = url.parse(value)
  if parsed_url.scheme and parsed_url.host then
    if not PROTOCOLS[parsed_url.scheme:upper()] then
      return false, "Supported protocols are: "..table.concat(PROTOCOLS, ", ")
    end
  end

  return true
end
```